heartbeat/monitors/factory.go

```diff
@@ -84,8 +92,22 @@ func newCommonPublishConfigs(info beat.Info, cfg *common.Config) (pipetool.Confi
 	}
 
 	var indexProcessor processors.Processor
+	if settings.DataStream != nil {
+		index := fmt.Sprintf("%s-%s-%s",
+			settings.DataStream.Type,
+			settings.DataStream.Dataset,
+			settings.DataStream.Namespace)
```

Comment:
if `Type` is not set you might want to set it to `synthetics`. Empty `Dataset` => `generic`, or empty `Namespace` would be `default`.

Reply:
Hmmmm, I actually wonder if we should just hardcode the `type` is there actually use-case for overriding it given that heartbeat only ever produces the one type?